### PR TITLE
Use project name as setting prefix (fixes #448)

### LINKED ISSUE #448: Settings names based on project name
One of the drawbacks of using cliquet in your app is that your settings names are prefixed with ``cliquet.`` (see [previous discussion](https://github.com/mozilla-services/cliquet/issues/318)).

Same for the management command, called ``cliquet`` to migrate ([see related kinto issue](https://github.com/Kinto/kinto/issues/129))

What if, in Cliquet, we would read the settings with a *dynamic* namespace based on the project name (this only one would be ``cliquet.project_name`` though) ?
Instead of having to define ``cliquet.http_host`` in the application, we would define ``readinglist.http_post``...

Didn't look about possible implementation so far. Just thinking loudly.

### PR DESCRIPTION
**Pull request by leplatrem**
* [x] Refactor usage of settings
* [ ] Document it

@ametaireau @Natim r?

### CODE REVIEW

File: cliquet/errors.py [hunk 1]
```diff
@@ -176,7 +176,7 @@ def send_alert(request, message=None, url=None, code='soft-eol'):
     :param url: The URL for more information, default to the documentation url.
     """
     if url is None:
-        url = request.registry.settings['cliquet.project_docs']
+        url = request.setting('project_docs')
```
Comment (ametaireau) [inline cliquet/errors.py]: `request.get_setting()`?

**Comment by ametaireau** [comment]
LGTM otherwise.

**Comment by ametaireau** [comment]
I actually wonder if that would make sense to store the settings without their prefix (at initialization time) so there is no need to ask for prefix + setting then?

**Comment by leplatrem** [comment]
> I actually wonder if that would make sense to store the settings without their prefix

Very good point indeed. Even though that represents a lot more work in the tests code... I'll have a look at what it could be. Especially because we already have some code that manipulates settings in `cliquet.initialize()`

File: cliquet/errors.py (hunk 1 above)
Comment (leplatrem) [inline cliquet/errors.py]: I liked it short, but I know what you mean :) This is indeed a broader question : #290

**Pull request closed without merging**